depends_on/common.py

```diff
@@ -286,23 +286,60 @@ def clone_repo(base_url, repo):
 def extract_gitlab_change(base_url, change_url, branch, main_branch, repo):
     "Extract the dependency by git cloning the repository in the right branch for the Merge Request."
     clone_repo(base_url, repo)
-    command(
-        f"cd {repo} && git remote add fork {change_url} && git fetch fork {branch} && git checkout -b mr/{branch} FETCH_HEAD"
-    )
+
+    add_fork_remote(repo, change_url)
+
+    cmd = f"""
+set -e
+cd {repo}
+git fetch fork {branch}
+git checkout -b mr/{branch} FETCH_HEAD
+"""
+
+    command(cmd.strip())
     merge_main_branch(repo, main_branch)
     return repo
 
 
 def extract_github_change(fork_url, branch, main_url, main_branch, repo):
     "Extract the dependency by git cloning the repository in the right branch for the Pull request."
     clone_repo(main_url, repo)
-    command(f"cd {repo} && git remote add fork {fork_url} && git fetch fork {branch}")
+    add_fork_remote(repo, fork_url)
+    cmd = f"""
+set -e
+cd {repo}
+git fetch fork {branch}
+"""
+
+    command(cmd.strip())
     # extract the master/main branch name
-    command(f"cd {repo} && git checkout -b pr/{branch} fork/{branch}")
+    cmd = f"""
+set -e
+cd {repo}
+git checkout -b pr/{branch} fork/{branch}
+"""
+    command(cmd.strip())
     merge_main_branch(repo, main_branch)
     return repo
 
 
+def add_fork_remote(repo, fork_url):
+    "Add a fork remote to a repository."
+
+    cmd = f"""
+set -e
+cd {repo}
+if git remote show fork &>/dev/null; then
```

Comment:
I suppose it could be better to remove it to support different forks and then to always create the fork.

Reply:
Done! Thanks!